enhancements/README.md

```diff
@@ -0,0 +1,23 @@
+# Enhancement Proposal
```

Comment:
We should add a link to this file in https://github.com/dexidp/dex/blob/master/.github/ISSUE_TEMPLATE/config.yml so users can find it when looking to open an issue.

Reply:
Also:
- Should we call these proposals DEPs (Dex Enhancement Proposals?)
- Should we setup a separate repository for them?
- Should we move the whole thing under `docs/` or some other more generic directory?
- How would EPs related to ADRs? Should we use both?

Reply:
Speaking about the issue template - no problem, I will add a link.

1. I'm good with it. It is more identical, and we can differentiate proposals for different projects in the future.
2. I think it would be best to start with this repo because now we do not have too many cases for writing proposals.  If it becomes inconvenient, we will always be able to move proposals to a separate repo.
3. This is a tricky question. As I see it, proposals are not connected to anything generic. That is why I moved them to the root. Although, it is discussable. If you think we should move them to docs, it is okay for me.
4. I assume we will have EPs for all complicated architecture choices in the future, so it is not necessary to use both.